bundles/framework/announcements/tool/AnnouncementsTool.js

```diff
@@ -0,0 +1,367 @@
+Oskari.clazz.define('Oskari.framework.announcements.tool.AnnouncementsTool',
+    function () {
+    }, {
+        index: 8,
+        pluginName: 'AnnouncementsPlugin',
+        allowedLocations: [],
+        allowedSiblings: [],
+        data: [],
+        selectedAnnouncements: [],
+        annTitles: [],
+        announcements: {},
+
+        isAnnouncementsDialogOpen: false,
+
+        groupedSiblings: true,
+
+        templates: {
+            announcements: jQuery('<div id="publisher-layout-announcements" class="tool-options">' + '<div id="publisher-layout-announcementsSelector">' + '<input type="text" name="publisher-announcements" id="publisher-announcements" disabled />' + '<button id="publisher-announcements-button"></button>' + '</div>' + '</div>'),
+            announcementsPopup: jQuery('<div id="publisher-announcements-popup">' + '<div id="publisher-announcements-inputs"></div>' + '</div>'),
+            inputCheckbox: jQuery('<div><input type="checkbox" /><label></label></div>')
+        },
+
+        eventHandlers: {
+            'Publisher2.ToolEnabledChangedEvent': function (event) {
+                var me = this;
+                var tool = event.getTool();
+                if (tool.getTool().id === me.getTool().id && tool.isStarted() && this.selectedAnnouncements) {
+                    me._sendAnnouncementsChangedEvent(this.selectedAnnouncements);
+                }
+            }
+        },
+
+        init: function (data) {
+            var me = this;
+            me.data = data;
+            me.selectedAnnouncements = [];
+            me.annTitles = [];
+
+            if (data.configuration && data.configuration.mapfull && data.configuration.mapfull.conf && data.configuration.mapfull.conf.plugins) {
+                _.each(data.configuration.mapfull.conf.plugins, function (plugin) {
+                    if (me.getTool().id === plugin.id) {
+                        me.setEnabled(true);
+                    }
+                });
```

Comment:
```suggestion
                const myId = this.getTool().id;
                me.setEnabled(data.configuration.mapfull.conf.plugins.some(plugin => myId === plugin.id));
```

Reply:
Just to be clear, there was nothing specifically wrong here. But we are trying to get rid of lodash since we now have most of the functions we use it for natively. Hence suggesting that we use native functions instead of an additional library when we don't need it. Sorry for not being clear on that before.